containerize/command.go

```diff
@@ -0,0 +1,261 @@
+package containerize
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"os/exec"
+	"runtime"
+	"sync"
+	"syscall"
+	"teleport-exec/cgroups"
+	"teleport-exec/filestream"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+//-------------------------------------------------------------------------------------------------
+type Command struct {
+	CommandId    string
+	Command      string
+	Args         []string
+	executor     *exec.Cmd
+	log          *ProcessLog
+	running      bool
+	failure      error
+	commandMutex sync.RWMutex
+	cgroup       *cgroups.Container
+}
+
+//-------------------------------------------------------------------------------------------------
+func NewCommand(command []string) Command {
+	return Command{
+		CommandId: uuid.NewString(),
+		Command:   command[0],
+		Args:      command[1:],
+		running:   false,
+	}
+}
+
+//-------------------------------------------------------------------------------------------------
+// Starts the command in a separate thread
+func (c *Command) Start() error {
+	if c.executor != nil {
```

Comment:
1. Why create the `exec.Command` here instead of in `NewCommand`? What happens if somebody creates a command with `NewCommand()`, doesn't `Start` it, and then calls `ResultCode`? (looks like it would panic)
2. This check should probably be done with the mutex held. Two concurrent calls to start could both see `executor` being nil and then sequentially try to start the command.

Reply:
Good call, did that and moved command creation into the constructor